.github/workflows/test-python.yml

```diff
@@ -7,18 +7,30 @@ jobs:
   run-unit-tests:
     
     name: Run all unit tests in codebase
-    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        platform: [ubuntu-latest, macos-latest]
+        python: [3.7]
+    runs-on: ${{ matrix.platform }}
     
     steps:
     - uses: actions/checkout@v2
-    - name: Set up Python 3.7
+    - name: Set up Python ${{ matrix.python }}
       uses: actions/setup-python@v2
       with:
-        python-version: 3.7.1
+        python-version: ${{ matrix.python }}
+
+    # - if: matrix.platform == 'windows-latest'
+    #   run: |
+    #     choco install wget --no-progress
+
+    - if: matrix.platform == 'macos-latest'
+      run: |
+        brew install opencv@2
```

Comment:
would prefer to use a miniconda CI rather than brew: https://github.com/borglab/gtsfm/blob/master/.github/workflows/test-python.yml